### ISSUE TITLE
Add support for ParU

### ISSUE THREAD

**Issue by mzy2240** [SciML/LinearSolve.jl#394]
Also just saw this news from Dr. Davis: https://github.com/DrTimothyAldenDavis/SuiteSparse/discussions/448. Might worth to add the support for ParU since UMFPACK is already supported.

_Originally posted by @mzy2240 in https://github.com/SciML/LinearSolve.jl/issues/392#issuecomment-1770979052_

**Comment by rayegun** [comment]
Just waiting on getting the actual build of ParU in Ygg before adding. Same with SPEX

**Comment by ChrisRackauckas** [comment]
Does such a jll exist now?

**Comment by mzy2240** [comment]
@rayegun just wondering any progress on ParU?

**Comment by rayegun** [comment]
Yes, I'll ping you when I polish it up and publish this weekend. ParU was unstable until recently

**Comment by mzy2240** [comment]
is https://github.com/JuliaBinaryWrappers/ParU_jll.jl rdy to use now? Seems still lack of major functions compared to KLU. Any example how to use it directly?
